src/paradigma/tremor/tremor_analysis.py

```diff
@@ -47,3 +49,57 @@ def extract_tremor_features_io(input_path: Union[str, Path], output_path: Union[
     write_df_data(metadata_time, metadata_samples, output_path, 'tremor_meta.json', df_windowed)
 
 
+def detect_tremor(df: pd.DataFrame, config: TremorDetectionConfig, path_to_classifier_input: Union[str, Path]) -> pd.DataFrame:
+   
+    # Initialize the classifier
+    coefficients = np.loadtxt(os.path.join(path_to_classifier_input, config.coefficients_file_name))
+    threshold = np.loadtxt(os.path.join(path_to_classifier_input, config.thresholds_file_name))
+
+    # Scale the mfcc's
+    mean_scaling = np.loadtxt(os.path.join(path_to_classifier_input, config.mean_scaling_file_name))
+    std_scaling = np.loadtxt(os.path.join(path_to_classifier_input, config.std_scaling_file_name))
+    mfcc = df.loc[:, df.columns.str.startswith('mfcc')]
+    mfcc_scaled = (mfcc-mean_scaling)/std_scaling
+
+    # Create a logistic regression model with pre-defined coefficients
+    log_reg = LogisticRegression(penalty = None)
+    log_reg.classes_ = np.array([0, 1]) 
+    log_reg.intercept_ = coefficients[0]
+    log_reg.coef_ = coefficients[1:].reshape(1, -1)
+    log_reg.n_features_in_ = int(mfcc.shape[1])
+    log_reg.feature_names_in_ = mfcc.columns
+
+    # Get the tremor probability 
+    df[DataColumns.PRED_TREMOR_PROBA] = log_reg.predict_proba(mfcc_scaled)[:, 1] 
+
+    # Make prediction based on pre-defined threshold
+    df[DataColumns.PRED_TREMOR_LOGREG] = df[DataColumns.PRED_TREMOR_PROBA] >= threshold
+    df[DataColumns.PRED_TREMOR_LOGREG] = df[DataColumns.PRED_TREMOR_LOGREG].astype(int) # save as int
```

Comment:
Dit kan in één regel: `df[DataColumn.PRED_TREMOR_LOGREG] = int(df[DataColumns.PRED_TREMOR_PROBA] >= threshold)` ofzo.